### PR TITLE
misc: drop dependency on prop-types and polyfill

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,7 +56,6 @@
     "moment": "2.19.4",
     "moment-duration-format": "2.2.2",
     "postis": "2.2.0",
-    "prop-types": "15.6.0",
```
Comment (virtuacoplenny) [inline package.json]: I guess it's sad atlaskit uses proptypes so it's still in package-lock.json.

File: react/features/base/dialog/components/web/StatelessDialog.js [hunk 2]
```diff
@@ -79,7 +78,7 @@ type Props = {
  * @type {ReactElement}
  */
 const ContextProvider = withContextFromProps({
-    i18n: PropTypes.object
+    i18n: Object
```
Comment (virtuacoplenny) [inline react/features/base/dialog/components/web/StatelessDialog.js]: I don't think this works as expected; at least it shows a warning for me locally. Seems like this should be a function that gets called that returns null or an Error. So something like
```
const ContextProvider = withContextFromProps({
    i18n: i18n => typeof i18n === 'object' ? null : new Error()
});
```
Essentially you'd do what prop-types is doing but you'd be doing it manually for this one instance.

File: package.json (hunk 1 above)
Comment (saghul) [reply package.json]: Indeed.

File: react/features/base/dialog/components/web/StatelessDialog.js (hunk 2 above)
Comment (saghul) [reply react/features/base/dialog/components/web/StatelessDialog.js]: I don't see any such warning. Not with eslint nor flow. Where are you seeing the problem?

**Comment by saghul** [comment]
@virtuacoplenny I think we hit the jackpot. Since the migration to React 16 looks like we can just drop atlaskit/layer-manager altogether and that context helper. PTAL.

File: react/features/base/dialog/components/web/StatelessDialog.js (hunk 2 above)
Comment (virtuacoplenny) [reply react/features/base/dialog/components/web/StatelessDialog.js]: I was seeing it at execution time when opening a modal.

**Comment by virtuacoplenny** [comment]
Yeah, looks like atlaskit dropped the layer manager requirement in 7.0.10 of modal dialog. I should have caught that. Thanks for cleaning up my stuff.

**Pull request merged**